plugins/modules/cloud_service.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Network to Code (@networktocode) <info@networktocode.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: cloud_service
+short_description: Creates or removes cloud_service from Nautobot
+description:
+  - Creates or removes cloud_service from Nautobot
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Travis Smith (@tsm1th)
+requirements:
+  - pynautobot
+version_added: "5.4.0"
+extends_documentation_fragment:
+  - networktocode.nautobot.fragments.base
+  - networktocode.nautobot.fragments.tags
+  - networktocode.nautobot.fragments.custom_fields
+options:
+  name:
+    description:
+      - The name of the cloud_service
+    required: true
+    type: str
+  description:
+    description:
+      - The description of the cloud_service
+    required: false
+    type: str
+  cloud_resource_type:
+    description:
+      - Required if I(state=present) and the cloud_service does not exist yet
+    required: false
+    type: raw
+  cloud_account:
+    description:
+      - A cloud account for this service.
+    required: false
+    type: raw
+"""
```

Comment:
Can we include `extra_config`?

Reply:
Added, but in the same situation as `config_schema` with idempotency.